src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceContainer/DiskSourceContainer.tsx

```diff
@@ -3,49 +3,109 @@ import { FieldPath, useFormContext } from 'react-hook-form';
 import { isUpstream } from '@kubevirt-utils/utils/utils';
 import { FormGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
 
-import { diskSourceEphemeralFieldID } from '../../utils/constants';
+import { REGISTRY_PASSWORD_FIELD, REGISTRY_USERNAME_FIELD } from '../../../utils/constants';
+import {
+  diskSourceEphemeralFieldID,
+  diskSourcePasswordFieldID,
+  diskSourceUsernameFieldID,
+} from '../../utils/constants';
 
 import { OS_REGISTERY_LINKS } from './utils/constants';
 
 type DiskSourceUrlInputProps = {
   fieldName: FieldPath<V1DiskFormState>;
+  isEphemeralDiskSource?: boolean;
   os: string;
 };
 
-const DiskSourceContainer: FC<DiskSourceUrlInputProps> = ({ fieldName, os }) => {
+const DiskSourceContainer: FC<DiskSourceUrlInputProps> = ({
+  fieldName,
+  isEphemeralDiskSource = false,
+  os,
+}) => {
   const { t } = useKubevirtTranslation();
-  const { getFieldState, register } = useFormContext<V1DiskFormState>();
+  const {
+    formState: { errors },
+    getFieldState,
+    register,
+  } = useFormContext<V1DiskFormState>();
+
   const isRHELOS = os?.includes(OS_NAME_TYPES.rhel);
-  // we show feodra on upstream and rhel on downstream, and default as fedora if not exists.
+  // we show fedora on upstream and rhel on downstream, and default as fedora if not exists.
   const exampleURL =
     isRHELOS && isUpstream
       ? OS_REGISTERY_LINKS.fedora
       : OS_REGISTERY_LINKS[os] || OS_REGISTERY_LINKS.fedora;
 
   const { error } = getFieldState(fieldName);
+
   return (
-    <FormGroup fieldId={diskSourceEphemeralFieldID} isRequired label={t('Container')}>
-      <TextInput
-        data-test-id={diskSourceEphemeralFieldID}
-        id={diskSourceEphemeralFieldID}
-        {...register(fieldName, { required: t('Container is required.') })}
-      />
-      <FormGroupHelperText validated={error ? ValidatedOptions.error : ValidatedOptions.default}>
-        {error ? (
-          error?.message
-        ) : (
-          <>
-            {t('Example: ')}
-            {exampleURL}
-          </>
-        )}
-      </FormGroupHelperText>
-    </FormGroup>
+    <>
+      <FormGroup fieldId={diskSourceEphemeralFieldID} isRequired label={t('Container')}>
+        <TextInput
+          data-test-id={diskSourceEphemeralFieldID}
+          id={diskSourceEphemeralFieldID}
+          {...register(fieldName, { required: t('Container is required.') })}
+        />
+        <FormGroupHelperText validated={error ? ValidatedOptions.error : ValidatedOptions.default}>
+          {error ? (
+            error?.message
+          ) : (
+            <>
+              {t('Example: ')}
+              {exampleURL}
+            </>
+          )}
+        </FormGroupHelperText>
+      </FormGroup>
+      {!isEphemeralDiskSource && (
+        <>
+          <FormGroup
+            className="disk-source-form-group"
+            fieldId={diskSourceUsernameFieldID}
+            label={t('Username')}
+          >
+            <FormTextInput
```

Comment:
Here you are using FormTextInput and above there's FormInput, what is the difference between them? can we use just one of them to remove one import?

Reply:
The comment in the FormTextInput file says: "PatternFly changes the signature of the 'onChange' handler for input elements. This causes issues with React Hook Form as it expects the default signature for an input element. So we have to create this wrapper component that takes care of converting these signatures for us."

Fixed!